scene/gui/label.cpp

```diff
@@ -367,10 +373,9 @@ void Label::_notification(int p_what) {
 			}
 
 			if (dirty || font_dirty || lines_dirty) {
-				_shape();
-				if (lines_dirty) {
+				if (!_shape()) {
```

Comment:
This change was the main one to get #73296 fixed, but that wouldn't be enough since it by itself would "unbalance" other parts of the algorithm, causing already fixed bugs to come back. That's why this PR had to go over the whole thing once again.